nautobot_netbox_importer/diffsync/nautobot.py

```diff
@@ -0,0 +1,362 @@
+
+        class_definition = {
+            "__annotations__": annotations,
+            "_attributes": attributes,
+            "_identifiers": identifiers,
+            "_modelname": self.content_type.replace(".", "_"),
+            "_wrapper": self,
+        }
+
+        meta = self.model._meta
+        if not meta.pk:
+            raise NotImplementedError("Missing primary key field")
+
+        annotations[self.pk_name] = INTERNAL_TYPE_TO_ANNOTATION[self.pk_type]
+        identifiers.append(self.pk_name)
+        class_definition[self.pk_name] = PydanticField()
+
+        for field_name, internal_type in self.fields.items():
+            if field_name in identifiers:
+                continue
+            if internal_type == "GenericForeignKey":
+                annotation = Tuple[ContentTypeStr, UUID]
+            elif internal_type in REFERENCE_INTERNAL_TYPES:
+                field = meta.get_field(field_name)
+                related_model = field.related_model
+                if not related_model:
+                    raise ValueError(f"Missing related model for field {field_name}")
+                referenced_internal_type = related_model._meta.pk.get_internal_type()
+                annotation = INTERNAL_TYPE_TO_ANNOTATION[referenced_internal_type]
+                if internal_type == "ManyToManyField":
+                    annotation = Set[annotation]
+            else:
+                annotation = INTERNAL_TYPE_TO_ANNOTATION[internal_type]
+
+            attributes.append(field_name)
+            annotations[field_name] = Optional[annotation]
+            class_definition[field_name] = PydanticField(default=None)
+
+        logger.debug("Creating model %s", class_definition)
+        try:
+            self.importer = type(class_definition["_modelname"], (ImporterModel,), class_definition)
+        except Exception:
+            logger.error("Failed to create model %s", class_definition, exc_info=True)
+            raise
+
+        return self.importer
+
+    def add_field(self, field_name: FieldName, internal_type: InternalFieldTypeStr) -> None:
+        """Add a field to the model."""
+        if self.importer:
+            raise RuntimeError("Cannot add fields after the importer has been created")
+
+        logger.debug("Adding nautobot field %s %s %s", self.content_type, field_name, internal_type)
+        if field_name in self.fields and self.fields[field_name] != internal_type:
+            raise ValueError(f"Field {field_name} already exists with different type {self.fields[field_name]}")
+        self.fields[field_name] = internal_type
+
+    def get_content_type(self) -> ContentType:
+        """Get the Nautobot content type ID for a given content type."""
+        return ContentType.objects.get_for_model(get_model_from_name(self.content_type))
+
+    def set_instance_defaults(self, **defaults: Any) -> None:
+        """Set default values for a Nautobot instance constructor."""
+        self.constructor_kwargs = defaults
```

Comment:
Would it make sense for this to be set at `__init__` time, or does it need to be a separate setter like this?

Reply:
The structure of wrappers is created on the fly. Some parts can be set in the constructors, however it's possible, that the wrapper will already exists, before configuring it.